core/app/services/spree/variants/remove_line_items.rb

```diff
@@ -0,0 +1,17 @@
+module Spree
+  module Variants
+    class RemoveLineItems
+      prepend Spree::ServiceModule::Base
+
+      def call(variant:)
+        cart_remove_item_service = Spree::Dependencies.cart_remove_item_service.constantize
+        incomplete_orders = variant.orders.where.not(state: 'complete')
+        incomplete_orders.each do |order|
```

Comment:
@aleksandarpetrushev in such cases when the list can be very big we should always use `find_each` as it's better for performance and memory management - https://apidock.com/rails/ActiveRecord/Batches/find_each